tests/test_image.py

```diff
@@ -1692,6 +1692,112 @@ def test_Image_binary_scalar_pow():
             image1 ** image2
 
 
+@timer
+def test_Image_inplace_setter():
+    """Test that all six types of supported Images set correctly
+    """
+    # Test that setting an Image initialized with an int array to a float array
+    # doesn't downcast and truncate values
+    farray = np.array([[1.2, 3.8], [5.6, 6.7]])
+    image = galsim.Image(np.zeros((2, 2), dtype=int))
+    image.array = farray
+    np.testing.assert_array_equal(image.array, farray)
```

Comment:
I'm not sure what the preference for broadcasting Image arrays is—this tests that an int array gets successfully cast to a float array, which may not be the desired behavior

Reply:
It is not desired behavior.